shared/src/main/scala/squants/Dimension.scala

```diff
@@ -84,6 +84,9 @@ trait Dimension[A <: Quantity[A]] {
   }
 
   implicit val dimensionImplicit: Dimension[A] = this
+
+  override def hashCode(): Int = getClass.getName.hashCode
```

Comment:
Isn't there some sort of rule that you should always override `hashCode` and `equals` together?

Reply:
The rule is that two values that are equal must have the same hashCode. This is already the case currently, but maybe I should enforce that?

Reply:
Right, we could make equals based on FQN or else would be based on object id, the latter should be stable I guess but maybe better to be consistent